### PR TITLE
[#9637] GET /course endpoint fails for courses where you're only a student or an instructor.

### LINKED ISSUE #9637: GET /course endpoint fails for courses where you're only a student or an instructor.
If you're both a student and an instructor, the GET /course endpoint will fail specifically for a course in which you're only a student of, or only an instructor of.

- **Environment**: `master` branch at commit 3455dda7e4e2d854471b1664548f76f002bfd4b9. 

**Steps to reproduce**
1. Login as instructor, create a course. Do not add yourself as a student into that course
2. Create a feedback session in the course
3. Try to edit the feedback session

**Expected behaviour**
All details of the feedback session loads.

**Actual behaviour**
None of the feedback session details load. 
image

**Additional info**

### PR DESCRIPTION
**Pull request by ChooJeremy**
Fixes #9637 

**Outline of Solution**

Just changed the access control to finish if student or instructor. Not sure if the tests to add the instructor as a student is the proper way to do it.

Ready for review.

### CODE REVIEW

File: src/main/java/teammates/ui/webapi/action/GetCourseAction.java [hunk 1]
```diff
@@ -19,22 +19,26 @@ protected AuthType getMinAuthLevel() {
...
     @Override
     public void checkSpecificAccessControl() {
-        if (!(userInfo.isInstructor || userInfo.isStudent)) {
-            throw new UnauthorizedAccessException("Student or instructor account is required to access this resource.");
-        }
-
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
 
-        if (userInfo.isInstructor) {
+        boolean checkForInstructorAccess = logic.getInstructorForGoogleId(courseId, userInfo.getId()) != null;
```
Comment (xpdavid) [inline src/main/java/teammates/ui/webapi/action/GetCourseAction.java]: Consider using `Const.ParamsNames.ENTITY_TYPE)`? What do you think?

I don't really like the idea of identifying "who you are" in the server side. It is expensive to do that. Instead, I would prefer the client (the frontend) to explicitly say the intent (e.g., indicate whether you are requesting the resource on behalf of student or instructor).

**Comment by ChooJeremy** [comment]
Updated to use EntityType. Makes it more consistent with the other APIs I think. This saves a database lookup when the user is a student and an instructor and is accessing a course in which he only has student access.

Tests for access control updated, frontend updated.

File: src/test/java/teammates/test/cases/webapi/GetCourseActionTest.java [hunk 2]
```diff
@@ -121,38 +125,93 @@ private void testNonExistentCourse() {
...
         verifyInaccessibleForAdmin(submissionParams);
+    }
+
+    @Test
+    protected void testAccessControl_onlyStudentOrInstructorAccessRequired_shouldPass()
+            throws InvalidParametersException, EnrollException, EntityDoesNotExistException,
+            EntityAlreadyExistsException {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        CourseAttributes typicalCourse2 = typicalBundle.courses.get("typicalCourse2");
+        logic.enrollStudents("Team|Section|Name|Email|Comment" + System.lineSeparator()
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/GetCourseActionTest.java]: I think you can just update a student's googleId to make it associated with the `instructor1OfCourse1.googleId`

File: src/web/services/course.service.ts [hunk 3]
```diff
@@ -16,11 +16,23 @@ export class CourseService {
   }
 
   /**
-   * Get course data by calling API.
+   * Get course data by calling API. Use as an instructor.
```
Comment (xpdavid) [inline src/web/services/course.service.ts]:
```suggestion
   * Get course data by calling API as an instructor.
```

File: src/web/services/course.service.ts [hunk 4]
```diff
@@ -16,11 +16,23 @@ export class CourseService {
...
+  getCourseAsInstructor(courseId: string): Observable<Course> {
     const paramMap: { [key: string]: string } = {
       courseid: courseId,
+      entitytype: 'instructor',
+    };
+    return this.httpRequestService.get('/course', paramMap);
+  }
+
+  /**
+   * Get course data by calling API. Use as a student.
```
Comment (xpdavid) [inline src/web/services/course.service.ts]:
```suggestion
   * Get course data by calling API as a student.
```

File: src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts [hunk 5]
```diff
@@ -182,7 +182,7 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
    */
   loadFeedbackSession(): void {
     // load the course of the feedback session first
-    this.courseService.getCourse(this.courseId).subscribe((course: Course) => {
+    this.courseService.getCourseAsInstructor(this.courseId).subscribe((course: Course) => {
```
Comment (xpdavid) [inline src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts]: Is this the only place where the API is used?

Comment (ChooJeremy) [reply src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts]: Yup. IntelliJ only reports this when I searched for usages. The students endpoint is still utilizing STUDENT_COURSE and STUDENT_COURSES

File: src/test/java/teammates/test/cases/webapi/GetCourseActionTest.java [hunk 6]
```diff
@@ -121,38 +124,95 @@ private void testNonExistentCourse() {
...
+        verifyAccessibleForStudentsOfTheSameCourse(submissionParams);
         verifyInaccessibleWithoutLogin(submissionParams);
         verifyInaccessibleForUnregisteredUsers(submissionParams);
+        verifyInaccessibleForInstructors(submissionParams);
         verifyInaccessibleForAdmin(submissionParams);
+    }
 
-        ______TS("instructors of the same course can access");
+    @Test
+    protected void testAccessControl_onlyStudentOrInstructorAccessRequired_shouldPass()
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/GetCourseActionTest.java]:
```suggestion
    protected void testAccessControl_loggedInEntityBothInstructorAndStudent_shouldBeAccessible()
```
better?

**Pull request merged**